### PR TITLE
[WIP] Try to fix inconsistent synchronization issues

### PR DESCRIPTION
**Pull request by tobiasreischmann**
I found some problems with the start dates of events and the setting of the event id, when updating the calendar. I have no idea if this might fix the problems of inconsistent calendar synchronization.

In my scenario, I have the problem that delegated calendars are not synchronized properly. Mainly, if I do not have full privileges on them. The changes of this PR at least lead to a case, where the events are temporarily displayed. Before that I was never able to see any of them.
However, I can not reproduce this state consistently. Sometime, I get a glimpse at the events, sometimes I don't. Sometimes I see the events for some seconds, then the calendar flickers and they are gone again...
I think this has to do with the fact that due to missing privileges I am not allowed to request the folder for the calendar from the server (see erGetFolder.js). This leads to the case, that this.supportsEvents in mivExchangeCalendar.js:2559 is false, which then refuses to return the cached events...
Help welcome!

### CODE REVIEW

File: calendar/interface/exchangeEvent/mivExchangeEvent.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ mivExchangeEvent.prototype = {
     },
 
     set startDate(aValue) {
-        if (aValue.toString() != this.startDate.toString()) {
+        if (aValue == null || this.startDate == null || aValue.toString() != this.startDate.toString()) {
```
Comment (tobiasreischmann) [inline calendar/interface/exchangeEvent/mivExchangeEvent.js]: I am not sure, if we should allow setting the value of null or prevent it. Currently however, this code leads to an exception, when setting null as startDate, which aborts the whole synchonization.

File: calendar/interface/exchangeCalendar/mivExchangeCalendar.js [hunk 2]
```diff
@@ -7475,7 +7475,7 @@ calExchangeCalendar.prototype = {
 
         //		item.setProperty("DESCRIPTION", aCalendarItem.getTagValue("t:Body"));
 
-        item.startDate = null;
+        // item.startDate = null;
```
[the remark this answers is not in the document]
Comment (advancingu) [reply calendar/interface/exchangeCalendar/mivExchangeCalendar.js]: Also, declare a temporary variable. Then you don't need to test `null` assignment (see below).

File: calendar/interface/exchangeEvent/mivExchangeEvent.js [hunk 3]
```diff
@@ -167,7 +167,7 @@ mivExchangeEvent.prototype = {
 
     set endDate(aValue) {
         //dump("set enddate: title:"+this.title+", aValue:"+aValue);
-        if (aValue.toString() != this.endDate.toString()) {
+        if (aValue == null || this.startDate == null || aValue.toString() != this.endDate.toString()) {
```
Comment (Zmegolaz) [inline calendar/interface/exchangeEvent/mivExchangeEvent.js]: I don't really know the flow here, but shouldn't it be `this.endDate`? Looks like a copy paste error from above.
```suggestion
        if (aValue == null || this.endDate == null || aValue.toString() != this.endDate.toString()) {
```

Comment (tobiasreischmann) [reply calendar/interface/exchangeEvent/mivExchangeEvent.js]: You are right. Thanks!

**Comment by advancingu** [comment]
Thanks a lot!

**Comment by tobiasreischmann** [comment]
Hey Markus, have you seen the WIP (work-in-progress) label and my comments? From my point of view I am unsure if these changes are correct and should be merged.
